content/source/docs/enterprise/vcs/troubleshooting.html.md

```diff
@@ -0,0 +1,66 @@
+- Bitbucket Server rejects the API request.
+
+To fix this issue, do the following:
+
+- Verify that the Docker instance running Terraform Enterprise can resolve the domain name and can reach Bitbucket Server.
+- Verify that the TFE client accepts the HTTPS connection to Bitbucket Server. This can be done by performing a `curl` from the TFE docker instance to Bitbucket Server; it should not return any SSL errors.
+- Verify that the Consumer Key, Consumer Name, and the Public Key is configured properly in Bitbucket Server.
+- Verify that the HTTP URL and API URL in TFE are correct for your Bitbucket Server instance. This includes the proper scheme (HTTP vs HTTPS), as well as the port.
+- Verify that the webhook URL is blank in Bitbucket Server's POST Receive Hooks Plugin
+
+## Bitbucket Cloud
+
+### Terraform Enterprise fails to obtain the list of repositories from Bitbucket Cloud.
+
+This typically happens when the TFE application in Bitbucket Cloud wasn't configured to have the full set of permissions.
+
+## GitHub
+
+### "Host key verification failed"" error in `terraform init` when attempting to ingress Terraform modules via Git over SSH
+
+This is most common when running Terraform 0.10.3 or 0.10.4, which had a bug in handling SSH submodule ingress. Try upgrading affected TFE workspaces to the latest Terraform version or 0.10.8 (the latest in the 0.10 series).
+
+### TFE can't ingress Git submodules, with auth errors during init
+
+This usually happens when an SSH key isn't associated with the VCS provider's OAuth client.
+
+- Go to your organization's "OAuth Configuration" settings page and check your GitHub client. If it still says "You can add a private SSH key to this connection to be used for git clone operations" (instead of "A private SSH key has been added..."), you need to click the "add a private SSH key" link and add a key.
+- Check the settings page for affected workspaces and ensure that "Include submodules on clone" is enabled.
+
+Note that the "SSH Key" section in a workspace's settings is only used for mid-run operations like cloning Terraform modules. It isn't used when cloning the configuration repo before a run.
+
+## General
+
+### Terraform Enterprise returns 500 after authenticating with the VCS provider (other than Bitbucket Server)
+
+The Callback URL in the OAuth application configuration in the VCS provider probably wasn't updated in the last step of the instructions and still points to the default "/" path (or an example.com link) instead of the full callback url.
+
+### Can't delete a workspace or module, with inscrutable 500 errors
+
+This often happens when the VCS connection has been somehow broken: it might have had permissions revoked, been reconfigured, or had the repository removed. Check for all three possibilities, take notes on what you find, and then contact HashiCorp support for further assistance.
```

Comment:
I'd replace "Check for..." with "Check for these possibilities, and contact HashiCorp support for further assistance, including any information you collected."